### DISCUSSION TITLE
[Q] My file is corrupted on Excel

### DISCUSSION THREAD

**Discussion by X0KCR3NnCNvn82kw** [exceljs/exceljs#2427] [Q&A]
Hello 👋 

I created a file with GoogleSheet that I then downloaded in xlsx format to use it with the library and save a new version of this file after modification. Once the new file is generated, I try to open it with Excel but I have errors and my style is not applied. At first I thought my code was probably bad so I just opened the file to create a new copy without modifying it and I still have its errors.

`node -e "const Excel = require('exceljs'); var workbook = new Excel.Workbook(); var data = workbook.xlsx.readFile('./template.xlsx').then(()=>workbook.xlsx.writeFile('./test.xlsx').then(()=>console.log('Finished writing')))"`

The file used: [template.xlsx](https://github.com/exceljs/exceljs/files/4980773/template.xlsx)

Errors:
`<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<recoveryLog xmlns="http://schemas.openxmlformats.org/spreadsheetml/2006/main">
    <logFileName>Repair Result to test1.xml</logFileName>
    <summary>Errors were detected in file 'test.xlsx'</summary>
    <removedParts summary="Following is a list of removed parts:">
        <removedPart>Removed Part: /xl/styles.xml part with XML error. (Styles) Load error.</removedPart>
    </removedParts>
    <repairedRecords summary="Following is a list of repairs:">
        <repairedRecord>Repaired Records: Cell information from /xl/worksheets/sheet1.xml part</repairedRecord>
        <repairedRecord>Repaired Records: Conditional formatting from /xl/worksheets/sheet1.xml part</repairedRecord>
    </repairedRecords>
</recoveryLog>`

Would the library have a problem or does the problem come from my template created with GoogleSheet?

Thanks 🙏 

**Config:**
MacOS v10.15.6
Node v10.20.1 (npm v6.14.4)
exceljs v4.1.1

**Comment by jagzviruz** [comment]
Having the same issue on 
exceljs v4.1.
node v14.5.0
Windows 10 - 1903

**Comment by Arlien** [comment]
+1

**Comment by AstroGD** [comment]
+1

**Comment by ywadi** [comment]
I had the same issue, and managed to pinpoint it but cant figure out how to fix it
The issue comes when you have conditional formatting, and the format/style for the condition has custom font color, or style. 
When I removed the font styles from the conditional formatting on my files, it worked. 
How can this be resolved is the question.

**Comment by ywadi** [comment]
> Hello 👋
> 
> I created a file with GoogleSheet that I then downloaded in xlsx format to use it with the library and save a new version of this file after modification. Once the new file is generated, I try to open it with Excel but I have errors and my style is not applied. At first I thought my code was probably bad so I just opened the file to create a new copy without modifying it and I still have its errors.
> 
> `node -e "const Excel = require('exceljs'); var workbook = new Excel.Workbook(); var data = workbook.xlsx.readFile('./template.xlsx').then(()=>workbook.xlsx.writeFile('./test.xlsx').then(()=>console.log('Finished writing')))"`
> 
> The file used: [template.xlsx](https://github.com/exceljs/exceljs/files/4980773/template.xlsx)
> 
> Errors:
> `<?xml version="1.0" encoding="UTF-8" standalone="yes"?> <recoveryLog xmlns="http://schemas.openxmlformats.org/spreadsheetml/2006/main"> <logFileName>Repair Result to test1.xml</logFileName> <summary>Errors were detected in file 'test.xlsx'</summary> <removedParts summary="Following is a list of removed parts:"> <removedPart>Removed Part: /xl/styles.xml part with XML error. (Styles) Load error.</removedPart> </removedParts> <repairedRecords summary="Following is a list of repairs:"> <repairedRecord>Repaired Records: Cell information from /xl/worksheets/sheet1.xml part</repairedRecord> <repairedRecord>Repaired Records: Conditional formatting from /xl/worksheets/sheet1.xml part</repairedRecord> </repairedRecords> </recoveryLog>`
> 
> Would the library have a problem or does the problem come from my template created with GoogleSheet?
> 
> Thanks 🙏
> 
> **Config:**
> MacOS v10.15.6
> Node v10.20.1 (npm v6.14.4)
> exceljs v4.1.1

Based on this, I removed the font styles from the template you attached and it worked. So thats where the issue is. Here is the fixed xlsx [https://gofile.io/d/5WbsQs](https://gofile.io/d/5WbsQs)
